pkg/controller/controller.go

```diff
@@ -289,7 +339,21 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		return nil, fmt.Errorf("claim Selector is not supported")
 	}
 
-	driverName, err := checkDriverState(p.grpcClient, p.timeout)
+	var checkSnapshotSupport bool
+	if options.PVC.Spec.DataSource != nil {
+		// PVC.Spec.DataSource.Name is the name of the VolumeSnapshot API object
+		if options.PVC.Spec.DataSource.Name == "" {
+			return nil, fmt.Errorf("the PVC source not found for PVC %s", options.PVC.Name)
+		}
+		if options.PVC.Spec.DataSource.Kind != snapshotKind {
+			return nil, fmt.Errorf("the PVC source is not the right type. Expected %s, Got %s", snapshotKind, options.PVC.Spec.DataSource.Kind)
+		}
+		if options.PVC.Spec.DataSource.APIGroup != snapshotAPIGroup {
+			return nil, fmt.Errorf("the PVC source does not belong to the right APIGroup. Expected %s, Got %s", snapshotAPIGroup, options.PVC.Spec.DataSource.APIGroup)
+		}
+		checkSnapshotSupport = true
+	}
+	driverName, err := checkDriverState(p.grpcClient, p.timeout, checkSnapshotSupport)
```

Comment:
Consider we will keep adding more types of data sources, a boolean value added here might not be convenient. But we could chance it later.

Reply:
Sure